tutorials/nservicebus-101/lesson-4/tutorial.md

```diff
@@ -0,0 +1,212 @@
+
+An event, on the other hand, is sent by one sender, and received by many receivers, or maybe one receiver, or even zero receivers.  This makes it an announcement that something has already happened. A subscriber can't reject or cancel an event any more than you could stop the New York Times from delivering newspapers to all of their subscribers. The publisher has no idea (and doesn't care) what receivers choose to do with the event; it's just making an announcement. So, events should be named in the [past tense](https://en.wikipedia.org/wiki/Past_tense), commonly ending with the **-ed** suffix, like `OrderPlaced`, `CustomerStatusUpdated`, and `CreditCardCharged`. This creates loose coupling, because while the contract (the content of the message) must be agreed upon, there is no requirement that subscribers of an event do anything.
+
+Let's take a look at all of these differences side-by-side:
+
+|   | Commands | Events |
+|---|:--------:|:------:|
+| Marker Interface | `ICommand` | `IEvent` |
+| Logical Senders | Many | 1 |
+| Logical Receivers | 1 | Many (or none) |
+| Purpose | "Please do something" | "Something has happened" |
+| Naming (Tense) | Imperative | Past |
+| Examples | `PlaceOrder`<br/> `UpdateCustomerStatus`<br/>`ChargeCreditCard` | `OrderPlaced`<br/>`CustomerStatusUpdated`<br/>`CreditCardCharged` |
+| Coupling Style | Tight | Loose |
+
+From this comparison, it's clear that commands and events will sometimes come in pairs. A command will arrive, perhaps from a website UI, telling the system to `DoSomething`. The system does that work, and as a result, publishes a `SomethingHappened` event, which other components in the system can react to.
+
+The loose coupling provided by publishing events gives us quite a bit of flexibility to design our software systems in a much more maintainable way.
+
+
+## Better code through decoupling
+
+Have you ever seen a codebase for a naive implementation of an e-commerce store? Commonly you'll find one gargantuan SubmitOrder method responsible for retrieving the shopping cart, creating an Order and OrderLines in the database, authorizing the credit card, capturing the credit card authorization, emailing a receipt, notifying a fulfillment agency, and then updating any sort of wish list, gift registry, or "frequently bought together" information the site might keep track of. If that method doesn't number in the hundreds of lines of code, then it likely calls into subroutine methods to handle each of these concerns. The sum total lines of code in such an example likely numbers in at least hundreds of lines of code, if not thousands.
+
+What happens when one of the steps in that long chain fails? You're left with a partially completed process that requires manual intervention to fix, either by mucking around manually in the database, manually reconciling with a credit card processor, or manually sending confirmation emails.
```

Comment:
When one of the steps in that long chain fails you're left with a partially completed process that requires manual intervention to fix, either by mucking around manually in the database, manually reconciling with a credit card processor, or manually sending confirmation emails.